File: pet-api/internal/models/Pet.go
package models

type Pet struct {
	PetId        int    `json:"pet_id"`
	PetName      string `json:"pet_name"`
	PetType      string `json:"pet_type"`
	PetBreed     string `json:"pet_breed"`
	PetColor     string `json:"pet_color"`
	PetBirthDate string `json:"pet_birth_date"`
	OwnerId      int    `json:"owner_id"`
}

File: pet-api/src/models/pet.rs
use chrono::NaiveDate;
use rust_decimal::Decimal;
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, sqlx::FromRow)]
pub struct PetModel {
    pub pet_id: String,
    pub pet_name: String,
    pub pet_birth_date: NaiveDate,
    pub pet_type: String,
    pub pet_breed: String,
    pub pet_weight: Decimal,
    pub pet_color: String,
    pub owner_id: String,
}

#[derive(Debug, Serialize, Deserialize, sqlx::FromRow)]
pub struct PetModelResponse {
    pub pet_id: String,
    pub pet_name: String,
    pub pet_birth_date: String,
    pub pet_type: String,
    pub pet_breed: String,
    pub pet_weight: Decimal,
    pub pet_color: String,
    pub owner_id: String,
}
